src/main/java/com/exasol/adapter/document/PipelineMonitor.java

```diff
@@ -0,0 +1,70 @@
+package com.exasol.adapter.document;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * This class monitors the fill state of the buffers in the {@link DataProcessingPipeline}.
+ */
+class PipelineMonitor extends Thread {
+    private static final Logger LOGGER = Logger.getLogger(PipelineMonitor.class.getName());
+    private final AtomicBoolean stopRequested = new AtomicBoolean(false);
+    long a = 0;
+    long b = 0;
+    long c = 0;
+    long d = 0;
+
+    /**
+     * Called when a row was loaded into the first buffer.
+     * 
+     * @param valueToPass a value
+     * @param <T>         type of the value
+     * @return value
+     */
+    public <T> T onEnterBuffer1(final T valueToPass) {
```

Comment:
Since this specific the data pipeline, would it make sense to rename buffers? For example, `schemaMappingBuffer` or `documentFetcherBuffer`?